pkg/controllers/jobflow/state/running.go

```diff
@@ -28,6 +28,8 @@ func (p *runningState) Execute(action v1alpha1.Action) error {
 		return SyncJobFlow(p.jobFlow, func(status *v1alpha1.JobFlowStatus, allJobList int) {
 			if len(status.CompletedJobs) == allJobList {
 				status.State.Phase = v1alpha1.Succeed
+			} else if len(status.FailedJobs) > 0 { // TODO(dongjiang199) Modify it when the if condition judgment is implemented
```

Comment:
What about checking the failed job is an expected behavior, for example there is a job depends on current job's failed condition, then we can better solve this controversial PR.

Reply:
Ensuring that the current jobflow FMS can be reversed is the expected behavior. 

However, there are also scenarios for conditional judgment based on error status; new features and state transition fixes can be implemented separately and step by step :)